Makefile

```diff
@@ -5,15 +5,19 @@ OBJCOPY := $(TARGET)-objcopy
 CFLAGS := -fPIC -O3 -fno-builtin-printf -fno-builtin-memcmp -nostdinc -nostdlib -nostartfiles -fvisibility=hidden -fdata-sections -ffunction-sections -I deps/secp256k1/src -I deps/secp256k1 -I deps/ckb-c-std-lib -I deps/ckb-c-std-lib/libc -I deps/ckb-c-std-lib/molecule -I c -I build -Wall -Werror -Wno-nonnull -Wno-nonnull-compare -Wno-unused-function -g
 LDFLAGS := -Wl,-static -fdata-sections -ffunction-sections -Wl,--gc-sections
 SECP256K1_SRC := deps/secp256k1/src/ecmult_static_pre_context.h
+# need difference include files (ckb_smt.h and moleculec2_reader.h)
+XUDT_RCE_CFLAGS := -fPIC -O3 -fno-builtin-printf -fno-builtin-memcmp -nostdinc -nostdlib -nostartfiles -fvisibility=hidden -fdata-sections -ffunction-sections -I deps/secp256k1/src -I deps/secp256k1 -I deps/ckb-c-stdlib-simulator-only2  -I deps/ckb-c-stdlib-simulator-only2/libc -I deps/ckb-c-stdlib-simulator-only2/molecule -I c -I build -Wall -Werror -Wno-nonnull -Wno-nonnull-compare -Wno-unused-function -g
```

Comment:
2 questions here:

* A huge portion of the config here is duplicate to `CFLAGS`, can we extract the common part, then define the difference as separate CFLAGS part?
* Here this is used to build xUDT script, however, the used dependency is named `ckb-c-stdlib-simulator-only2`, I'm confused here, if it is actually used to build the script, it shall not be call simulator only.

Reply:
I'm not sure we can upgrade current `ckb-c-stdlib`, since it might be used by other existing scripts such as `simple_udt.c`, or `anyone_can_pay.c`, we will need a new checkout of the repository

Reply:
I don't think either `simple_udt.c` or `anyone_can_pay.c` uses `ckb_dlfcn.h`.

Reply:
One different way, is that you can try updating `ckb-c-stdlib`, if the binary hash stays the same, we can safely upgrade, otherwise we will need to maintain different versions of `ckb-c-stdlib`